File: Lesson09_OOP/07_Tea/TeaPot.java
/*
Copyright Ann Barcomb and Emily Marasco, 2022
Licensed under GPL v3
See LICENSE.txt for more information.
*/

public class TeaPot {
    private final Teabag bag;
    private boolean water = false;
    private boolean teabagIn = true;
       
    public void printWaterStatus() {
        if (this.water == true) {
            System.out.println("Water has been added to the teapot.");
        } else {
            System.out.println("Water has not been added to the teapot.");
        }
    }

    public void printTeabagStatus() {
        if (this.teabagIn == false) {
            System.out.println("The teabag has been removed.");
        } else {
            System.out.println("The teabag is still in the teapot.");
        }
    }

    public void printTeabagDetails() {
        if (this.teabagIn == false) {
            System.out.println("The teabag has been removed and thrown away.");
        } else {
            System.out.printf("We are brewing %s, from %s, by %s.\n",
                this.bag.getTeaType(), this.bag.getTeaOrigin(),
                this.bag.getTeaBrand());
        }
    }

    public void removeTeabag() {
        this.teabagIn = false;
    }

    public void addWater() {
        this.water = true;
    }

    public TeaPot() {
        this.bag = new Teabag();
    }

    public TeaPot(String teaType) {
        this.bag = new Teabag(teaType);
    }

    public TeaPot(String teaType, String origin) {
        this.bag = new Teabag(teaType, origin);
    }

    public TeaPot(String teaType, String origin, String brand) {
        this.bag = new Teabag(teaType, origin, brand);
    }
}

File: Lesson09_OOP/07_Tea/teapot.py
"""
Copyright Khawla Shnaikat and ChatGPT 3.5, 2023
Licensed under GPL v3
See LICENSE.txt for more information.
"""
from teabag import Teabag
class TeaPot:
    def __init__(self, tea_type="samdayeon", tea_origin="Korea", tea_brand="unknown"):
        self.bag = Teabag(tea_type, tea_origin, tea_brand)
        self.water = False
        self.teabag_in = True

    def print_water_status(self):
        if self.water:
            print("Water has been added to the teapot.")
        else:
            print("Water has not been added to the teapot.")

    def print_teabag_status(self):
        if not self.teabag_in:
            print("The teabag has been removed.")
        else:
            print("The teabag is still in the teapot.")

    def print_teabag_details(self):
        if not self.teabag_in:
            print("The teabag has been removed and thrown away.")
        else:
            print(f"We are brewing {self.bag.get_tea_type()}, from {self.bag.get_tea_origin()}, by {self.bag.get_tea_brand()}.")

    def remove_teabag(self):
        self.teabag_in = False

    def add_water(self):
        self.water = True
